src/main/cpp/src/utils/vid_mapper_sql.cc

```diff
@@ -0,0 +1,292 @@
+    auto& ref = m_field_idx_to_info[field_idx];
+
+    const char* c_field_type = dbi_result_get_string(result, DBTABLE_FIELD_COLUMN_TYPE.c_str());
+    std::string field_type = ((NULL == c_field_type) ? "" : c_field_type);
+    if (! field_type.empty()) {
+      auto iter = VidMapper::m_typename_string_to_type_index.find(field_type);
+      VERIFY_OR_THROW(iter != VidMapper::m_typename_string_to_type_index.end()
+          && "Field type not handled");
+      ref.m_type_index = (*iter).second;
+    }
+    if (! field_type.empty()) {
+      auto iter = VidMapper::m_typename_string_to_bcf_ht_type.find(field_type);
+      VERIFY_OR_THROW(iter != VidMapper::m_typename_string_to_bcf_ht_type.end()
+          && "Field type not handled");
+      ref.m_bcf_ht_type = (*iter).second;
+    }
+
+    signed char is_filter = dbi_result_get_char(result, DBTABLE_FIELD_COLUMN_FILTER.c_str());
+    ref.m_is_vcf_FILTER_field = ((is_filter == 't') ? true : false);
+
+    signed char is_format = dbi_result_get_char(result, DBTABLE_FIELD_COLUMN_FORMAT.c_str());
+    ref.m_is_vcf_FORMAT_field = ((is_format == 't') ? true : false);
+
+    signed char is_info = dbi_result_get_char(result, DBTABLE_FIELD_COLUMN_INFO.c_str());
+    ref.m_is_vcf_INFO_field = ((is_info == 't') ? true : false);
+
+    const char* c_length_type = dbi_result_get_string(result, DBTABLE_FIELD_COLUMN_LENTYPE.c_str());
+    std::string length_type = ((NULL == c_length_type) ? "" : c_length_type);
+    auto known_field_enum = 0u;
+    auto is_known_field = KnownFieldInfo::get_known_field_enum_for_name(field_name, known_field_enum);
+
+    if ((0 == length_type.compare("ERROR")) || length_type.empty()) {
+      if (is_known_field) {
+        auto length_descriptor = KnownFieldInfo::get_length_descriptor_for_known_field_enum(known_field_enum);
+        ref.m_length_descriptor = length_descriptor;
+        if (length_descriptor == BCF_VL_FIXED) {
+          ref.m_num_elements = KnownFieldInfo::get_num_elements_for_known_field_enum(known_field_enum, 0u, 0u);
+        }
+      } else {
+        ref.m_num_elements = 1;
+        ref.m_length_descriptor = BCF_VL_FIXED;
+      }
+    } else {
+      if (0 == length_type.compare("NUM")) {
+        ref.m_length_descriptor = BCF_VL_FIXED;
+        ref.m_num_elements = dbi_result_get_int(result, DBTABLE_FIELD_COLUMN_LENVAL.c_str());
+      } else {
+        if (0 == length_type.compare("A")) {
+            ref.m_length_descriptor = BCF_VL_A;
+        } else if (0 == length_type.compare("G")) {
+            ref.m_length_descriptor = BCF_VL_G;
+        } else if (0 == length_type.compare("R")) {
+            ref.m_length_descriptor = BCF_VL_R;
+        } else {
+          if (0 == length_type.compare("VAR")) {
+              ref.m_length_descriptor = BCF_VL_VAR;
+          }
+        }
+      }
+    }
+
+    std::string combine_op = dbi_result_get_string(result, DBTABLE_FIELD_COLUMN_COMBOP.c_str());
```

Comment:
Can the result be null? Based on your PR https://github.com/Intel-HLS/GenomicsSampleAPIs/pull/33, it looks like it can be